docfx_yaml/extension.py

```diff
@@ -365,19 +382,133 @@ def _update_friendly_package_name(path):
         'langs': ['python'],
     }
 
-    # Only add summary to parts of the code that we don't get it from the monkeypatch
-    if _type == MODULE:
+    summary_info = {
+        'variables': {},  # Stores mapping of variables and its description, types
+        'returns': [],    # Stores the return info
+        'exceptions': []  # Stores the exception info
+    }
+
+    # Add extracted summary
```

Comment:
Is there any of this we can pull out into separate functions? The indentation gets pretty deep here, making it more difficult to follow.